src/widgets/reflow.rs

```diff
@@ -72,155 +72,142 @@ where
             return None;
         }
 
-        let mut current_line: Option<Vec<StyledGrapheme<'a>>> = None;
-        let mut line_width: u16 = 0;
-
-        // Try to repeatedly retrieve next line
-        while current_line.is_none() {
-            // Retrieve next preprocessed wrapped line
-            if let Some(line_iterator) = &mut self.wrapped_lines {
-                if let Some(line) = line_iterator.next() {
-                    line_width = line
-                        .iter()
-                        .map(|grapheme| grapheme.symbol.width())
-                        .sum::<usize>() as u16;
-                    current_line = Some(line);
-                }
+        loop {
+            // emit next preprocessed line if present
+            'emit: {
+                let Some(line_iterator) = &mut self.wrapped_lines else {
+                    break 'emit;
+                };
+                let Some(line) = line_iterator.next() else {
+                    break 'emit;
+                };
+
+                let line_width = line
+                    .iter()
+                    .map(|grapheme| grapheme.symbol.width() as u16)
+                    .sum();
+
+                self.current_line = line;
+                return Some(WrappedLine {
+                    line: &self.current_line,
+                    width: line_width,
+                    alignment: self.current_alignment,
+                });
             }
```

Comment:
Unfortunately I can only extract the first half of the short-circuit logic, because the borrow checker bends under the weight of borrowing other parts of `Self`, like `self.current_alignment`.

I still see this as a net gain though - the let-elses + `'emit` are too clunky for my taste after all.